default-fields/0000-default-fields.md

```diff
@@ -11,14 +11,182 @@ One para explanation of the feature.
+struct Foo {
+    a: &'static str,
+    b: bool,
+    c: i32
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+    ..Default::default()
+}
+```
+
+The problem with this is that it requires Foo to derive Default, which may not always
+be desirable.
+
+The existing struct construction also can't be used when a struct has private fields:
+
+```rust
+mod data {
+    pub struct Foo {
+        pub a: &'static str,
+        b: bool,
+        pub c: i32
+    }
+}
+
+// Error: field `b` is private
+let foo = data::Foo {
+    a: "Hello",
+    b: true,
+    c: 42
+}
+```
+
+To work around these shortcomings, users can create constructor functions or more elaborate builders.
+The problem with a constructor is that it needs to be kept in sync with the fields of the struct, or
+require the user reassign values they want to change.
+Builders enable more advanced construction, but also need to kept in-line with the struct they build,
+and can be overkill for large, but uncomplicated structs.
+
+An alternative solution is to allow struct definitions to supply default values for individual fields,
+and then allow those fields to be omitted from construction:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42
+}
+```
+
+The goal of this syntax is to let users build a struct as if it default fields weren't there.
 
 # Detailed design
 [design]: #detailed-design
 
-This is the bulk of the RFC. Explain the design in enough detail for somebody familiar
-with the language to understand, and for somebody familiar with the compiler to implement.
-This should get into specifics and corner-cases, and include examples of how the feature is used.
+With field defaults a caller can still instantiate this struct without needing builders
+or a constructor function:
+
+```rust
+mod data {
+    pub struct Foo {
+        pub a: &'static str,
+        b: bool = true,
+        pub c: i32
+    }
+}
+
+// Error: field `b` is private
+let foo = data::Foo {
+    a: "Hello",
+    c: 42
+}
+```
+
+This also means additional fields can be added to a struct in a non-breaking fashion.
+Say we have the following API and consumer:
+
+```rust
+mod data {
+    pub struct Foo {
+        pub a: &'static str,
+        pub c: i32
+    }
+}
+
+let foo = data::Foo {
+    a: "Hello",
+    c: 42
+}
+```
+
+We can add a new field b to this struct with a default value, and the calling code
+doesn't need to change:
+
+```rust
+mod data {
+    pub struct Foo {
+        pub a: &'static str,
+        pub b: bool = true
+        pub c: i32
+    }
+}
+
+let foo = data::Foo {
+    a: "Hello",
+    c: 42
+}
+```
+
+This is the key use-case for avoiding having to use special syntax like `..` to utilise
+default fields.
+If it's unclear whether or not a particular caller will use a default field value then
+its addition can't be treated as a non-breaking change.
+
+The semantics of field defaults are the same as consts.
```

Comment:
Happy to be convinced otherwise on this one, but following the same semantics as `const` seemed like a good starting point.